pkg/api/api.go

```diff
@@ -134,14 +137,53 @@ func GetSecrets(context APIContext, lastETag string, project string, config stri
 	if config != "" {
 		params = append(params, QueryParam{Key: "config", Value: config})
 	}
+	if nameTransformer != "" {
+		params = append(params, QueryParam{Key: "name_transformer", Value: nameTransformer})
+	}
+	if format != "" {
+		params = append(params, QueryParam{Key: "format", Value: format})
+	}
 
 	response, err := GetRequest(context, "/v3/configs/config/secrets/download", headers, params)
 	if err != nil {
 		return nil, err
 	}
-	result, modelErr := models.ParseSecrets(response.HTTPResponse.StatusCode, response.Body, response.HTTPResponse.Header.Get("ETag"))
+
+	if response.HTTPResponse.StatusCode == 304 {
+		return &models.SecretsResult{Modified: false, Secrets: nil, ETag: ""}, nil
+	}
+	eTag := response.HTTPResponse.Header.Get("ETag")
+
+	// Format defeats JSON parsing
+	if format != "" {
+		secrets := []models.Secret{{
+			Name:  secretsDownloadFileKey,
+			Value: string(response.Body),
+		}}
+		return &models.SecretsResult{Modified: true, Secrets: secrets, ETag: eTag}, nil
+	}
+
+	result, modelErr := parseSecrets(response.Body, eTag)
 	if modelErr != nil {
 		return nil, &APIError{Err: modelErr, Message: "Unable to parse secrets"}
 	}
 	return result, nil
 }
+
+func parseSecrets(response []byte, eTag string) (*models.SecretsResult, error) {
+	var result map[string]string
+	err := json.Unmarshal(response, &result)
+	if err != nil {
+		return nil, err
+	}
+
+	secrets := make([]models.Secret, 0)
+	for key, value := range result {
+		secret := models.Secret{Name: key, Value: value}
+		secrets = append(secrets, secret)
+	}
+	sort.Slice(secrets, func(i, j int) bool {
+		return secrets[i].Name < secrets[j].Name
+	})
+	return &models.SecretsResult{Modified: true, Secrets: secrets, ETag: eTag}, nil
+}
```

Comment:
@nmanoogian I brought `parseSecrets` here because it's not being used by any other package.